api/event_stream_test.go

```diff
@@ -302,6 +306,19 @@ func TestEventStream_PayloadValueHelpers(t *testing.T) {
 				}, n)
 			},
 		},
+		{
+			desc:  "node_pool",
+			input: []byte(`{"Topic":"NodePool","Payload":{"NodePool":{"Description":"prod pool","Name":"prod"}}}`),
+			expectFn: func(t *testing.T, event Event) {
+				must.Eq(t, TopicNodePool, event.Topic)
+				n, err := event.NodePool()
+				must.NoError(t, err)
+				must.Eq(t, &NodePool{
+					Name:        "prod",
+					Description: "prod pool",
+				}, n)
+			},
+		},
```

Comment:
This new test case is fine but this test sure is weird to me. Do you know why we're testing that we can unmarshal the event JSON to structs? I feel like I'm missing something :grinning:

Reply:
Hum...right, that's kind of weird 😅 

I guess it's testing the `event.NodePool()` method that extracts a `NodePool`object from the event? But that is also mostly just mapstructure 🤷